pkg/cmd/infra/deployer/deployer.go

```diff
@@ -77,88 +74,111 @@ func NewCommandDeployer(name string) *cobra.Command {
 	return cmd
 }
 
+func NewDeployer(client kclient.Interface) *Deployer {
+	return &Deployer{
+		getDeployment: func(namespace, name string) (*kapi.ReplicationController, error) {
+			return client.ReplicationControllers(namespace).Get(name)
+		},
+		getControllers: func(namespace string) (*kapi.ReplicationControllerList, error) {
+			return client.ReplicationControllers(namespace).List(labels.Everything())
+		},
+		updateDeployment: func(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
+			return client.ReplicationControllers(namespace).Update(deployment)
+		},
+		strategyFor: func(config *deployapi.DeploymentConfig) (strategy.DeploymentStrategy, error) {
+			switch config.Template.Strategy.Type {
+			case deployapi.DeploymentStrategyTypeRecreate:
+				return recreate.NewRecreateDeploymentStrategy(client, latest.Codec), nil
+			case deployapi.DeploymentStrategyTypeRolling:
+				recreate := recreate.NewRecreateDeploymentStrategy(client, latest.Codec)
+				return rolling.NewRollingDeploymentStrategy(config.Namespace, client, latest.Codec, recreate), nil
+			default:
+				return nil, fmt.Errorf("unsupported strategy type: %s", config.Template.Strategy.Type)
+			}
+		},
+	}
+}
+
+type Deployer struct {
+	strategyFor      func(config *deployapi.DeploymentConfig) (strategy.DeploymentStrategy, error)
+	getDeployment    func(namespace, name string) (*kapi.ReplicationController, error)
+	getControllers   func(namespace string) (*kapi.ReplicationControllerList, error)
+	updateDeployment func(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error)
+}
+
 // deploy executes a deployment strategy.
```

Comment:
I'll be doing a godocs sweep after I finish the unit tests, there's a lot to do.